include/dlaf/matrix/panel.h

```diff
@@ -442,23 +438,139 @@ struct Panel<axis, const T, D> {
-struct Panel : public Panel<axis, const T, D> {
+struct Panel<axis, const T, D, StoreTransposed::Yes>
+    : protected Panel<orthogonal(axis), const T, D, StoreTransposed::No> {
+private:
+  using BaseT = Panel<orthogonal(axis), const T, D, StoreTransposed::No>;
+
+  static matrix::Distribution transposeDist(const matrix::Distribution& dist) {
+    return {common::transposed(dist.size()), common::transposed(dist.blockSize()),
+            common::transposed(dist.commGridSize()), common::transposed(dist.rankIndex()),
+            common::transposed(dist.sourceRankIndex())};
+  }
+
+public:
   constexpr static Coord coord = orthogonal(axis);
   constexpr static Device device = D;
 
   using TileType = Tile<T, D>;
   using ConstTileType = Tile<const T, D>;
   using ElementType = T;
 
-  explicit Panel(matrix::Distribution distribution, GlobalTileIndex start = {0, 0})
-      : Panel<axis, const T, D>(std::move(distribution), std::move(start)) {}
+  Panel(matrix::Distribution distribution, GlobalTileIndex start = {0, 0})
+      : BaseT(transposeDist(distribution), common::transposed(start)),
+        dist_parent_(std::move(distribution)) {}
+
+  Distribution parentDistribution() const {
+    return dist_parent_;
+  }
+
+  auto iteratorLocal() const {
+    return common::iterate_range2d(LocalTileIndex(coord, BaseT::rangeStartLocal()),
+                                   LocalTileIndex(coord, BaseT::rangeEndLocal(), 1));
+  }
+
+  void setRange(GlobalTileIndex start_idx, GlobalTileIndex end_idx) {
+    start_idx.transpose();
+    end_idx.transpose();
+    BaseT::setRange(start_idx, end_idx);
+  }
+
+  void setRangeStart(GlobalElementIndex index) {
+    index.transpose();
+    BaseT::setRangeStart(index);
+  }
+
+  void setRangeStart(GlobalTileIndex index) {
+    index.transpose();
+    BaseT::setRangeStart(index);
+  }
+
+  void setRangeEnd(GlobalTileIndex end_idx) {
+    end_idx.transpose();
+    BaseT::setRangeEnd(end_idx);
+  }
+
+  using BaseT::rangeStart;
+  using BaseT::rangeStartLocal;
+
+  using BaseT::rangeEnd;
+  using BaseT::rangeEndLocal;
+
+  // TODO fix setHeight (and setWidth accordingly)
+  template <Coord A = axis, std::enable_if_t<A == axis && Coord::Row == axis, int> = 0>
+  void setHeight(const SizeType height) {
+    BaseT::setWidth(height);
+  }
+
+  // TODO fix setHeight (and setWidth accordingly)
+  template <Coord A = axis, std::enable_if_t<A == axis && Coord::Col == axis, int> = 0>
+  void setWidth(const SizeType width) {
+    BaseT::setHeight(width);
+  }
+
+  // TODO fix setHeight (and setWidth accordingly)
+  template <Coord A = axis, std::enable_if_t<A == axis && Coord::Row == axis, int> = 0>
+  SizeType getHeight() const noexcept {
+    return BaseT::getWidth();
+  }
+
+  // TODO fix setHeight (and setWidth accordingly)
+  template <Coord A = axis, std::enable_if_t<A == axis && Coord::Col == axis, int> = 0>
+  SizeType getWidth() const noexcept {
+    return BaseT::getHeight();
+  }
```

Comment:
Width/Height are maintained as they were originally also for `StoreTransposed::Yes`, so the mutable size is:
- "width" in a Col panel
- "height" in a Row panel

To me it sounds reasonable, but before removing the TODO I wait for your feedbacks.